From 55e0f262c77a5f9f2d2d6b833a58b9293efcdc4b Mon Sep 17 00:00:00 2001
From: caodongxia <315816521@qq.com>
Date: Wed, 11 Aug 2021 10:43:22 +0800
Subject: [PATCH] fix algparse unknown option -d

---
 programs/algparse/algparse.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/programs/algparse/algparse.c b/programs/algparse/algparse.c
index 3820d63..afd5702 100644
--- a/programs/algparse/algparse.c
+++ b/programs/algparse/algparse.c
@@ -566,7 +566,8 @@ static void usage(void)
 		"    -P <password> | -nsspw <password> | -password <password>:\n"
 		"        <password> to unlock crypto database\n"
 		"    -v --verbose: be more verbose\n"
-		"    -d --debug: enable debug logging\n"
+		"    --debug: enable debug logging\n"
+                /* -d <NSSDB> is reserved */
 		"    --impair: disable all algorithm parser checks\n"
 		"    --ignore: ignore parser errors (or at least some)\n"
 		"    -p1: simple parser\n"
-- 
2.27.0

