diff --git a/programs/pluto/security_selinux.c b/programs/pluto/security_selinux.c
index ac5028e..2073a4a 100644
--- a/programs/pluto/security_selinux.c
+++ b/programs/pluto/security_selinux.c
@@ -18,6 +18,8 @@
 
 static int selinux_ready = 0;
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 void init_avc(void)
 {
 	if (!is_selinux_enabled()) {
@@ -82,3 +84,4 @@ int within_range(security_context_t sl, security_context_t range)
 		range);
 	return 1;
 }
+#pragma GCC diagnostic pop
diff --git a/programs/pluto/security_selinux.h b/programs/pluto/security_selinux.h
index cccd60f..61279eb 100644
--- a/programs/pluto/security_selinux.h
+++ b/programs/pluto/security_selinux.h
@@ -20,6 +20,9 @@
 #include <selinux/context.h>
 
 void init_avc(void);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 int within_range(security_context_t sl, security_context_t range);
+#pragma GCC diagnostic pop
 
 #endif /* _SECURITY_SELINUX_H */
